### PR TITLE
[WIP]: Add storybook

### PR DESCRIPTION
**Pull request by FilipMessa**
**Type:** Feature

The following has been addressed in the PR:

**Resolves**  #140 

**Description:**
Basic Storybook setup with first stories for Icon's component.
I plan to deliver other stories in the next PRs because I want to keep this PR small. And also I would like to add regression tests later.

Please check if this implementation is OK for you. Thanks and have a nice day.

### CODE REVIEW

**Comment by FilipMessa** [comment]
Looks like the issue is with the `./muiComponents` types but I didn't touch them. 
So probably `yarn` update some module and then these type errors appeared. 
🤔

**Comment by FilipMessa** [comment]
I replaced the `Button from '../../muiComponents/Button'` with `import Button from '@material-ui/core/Button';` and lint issue for the Button disappears, so I look like that have something with how the `forwardRef` type is added to components.

File: src/components/Icon/Icon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
Comment (sergiohgz) [inline src/components/Icon/Icon.stories.tsx]: We should talk about folder structure. Like snapshots, I prefer having a folder called __stories__, but in that case, we have to refactor tests to put them in their specific folders (e.g, __tests__)

Comment (FilipMessa) [reply src/components/Icon/Icon.stories.tsx]: Do you mean something like? 

```
src/
│ 
│─── components
   │─── Icon
      │─── stories
      │─── tests
         │─── __snapshots__
```

Comment (sergiohgz) [reply src/components/Icon/Icon.stories.tsx]: Something like that, but with a unification of both folders (tests and stories), prefixing and suffixing using double underscores. What do you think about?
Also, I would like to hear @priscilawebdev and @juanpicado (but this talk could be extracted to an issue and discuss there :wink: )

**Comment by priscilawebdev** [comment]
A design system was part of my plans for the Verdaccio UI. I'm glad you already started it. Thank you @FilipMessa  👏🙂

File: src/components/Icon/Icon.stories.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/components/Icon/Icon.stories.tsx]: thank you for asking @sergiohgz ...Hmm 🧐, I liked the @FilipMessa approach. I like to always keep tests together with it's component/container .. what do you think @juanpicado?

Comment (juanpicado) [reply src/components/Icon/Icon.stories.tsx]: Agree with @priscilawebdev

Comment (juanpicado) [reply src/components/Icon/Icon.stories.tsx]: For more context, usuall the `__xxx__` are intended for tests or fixtures. Stories are non both of them, but, let's experiment for now, I think should be fine. If we realize is not a good approach, we can change it in the future.

Comment (sergiohgz) [reply src/components/Icon/Icon.stories.tsx]: Let's keep as it's now for simplicity. We can discuss later if folders contains too much things :wink:

**Comment by juanpicado** [comment]
@priscilawebdev this sounds familiar to you? It does not happen on master 🤔 
```
src/muiComponents/Tooltip/Tooltip.tsx:8:11 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: TooltipProps, context?: any): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | Component<...> | null', gave the following error.
    Type '{ innerRef: Ref<HTMLDivElement>; children: ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | ... 5 more ... | (ReactElement<...> & ReactPortal); ... 269 more ...; ref?: ((instance: unknown) =>...' is not assignable to type 'IntrinsicAttributes'.
      Types of property 'css' are incompatible.
        Type 'Interpolation' is not assignable to type 'InterpolationWithTheme<any>'.
          Type 'TemplateStringsArray' is not assignable to type 'InterpolationWithTheme<any>'.
            Type 'TemplateStringsArray' is not assignable to type 'ObjectInterpolation<undefined>'.
  Overload 2 of 2, '(props: PropsWithChildren<TooltipProps>, context?: any): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | Component<...> | null', gave the following error.
    Type '{ innerRef: Ref<HTMLDivElement>; children: ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | ... 5 more ... | (ReactElement<...> & ReactPortal); ... 269 more ...; ref?: ((instance: unknown) =>...' is not assignable to type 'IntrinsicAttributes'.
      Types of property 'css' are incompatible.
        Type 'Interpolation' is not assignable to type 'InterpolationWithTheme<any>'.
          Type 'TemplateStringsArray' is not assignable to type 'InterpolationWithTheme<any>'.
            Type 'TemplateStringsArray' is not assignable to type 'ObjectInterpolation<undefined>'.

8   return <MaterialUITooltip {...props} innerRef={ref} />;
```

**Comment by tmkn** [comment]
@juanpicado This PR compiles fine for me, also when merging with the current master, idk why those errors are there. I guess stale data within the CI?

**Comment by sergiohgz** [comment]
Looking at CI, there are linting and testing problems. Those problems don't occurs on current master. Did you rebase and introduce a problem with conflicts?

Note: I tried to merge current master into your branch and the problems still happens on my local, so something has been broken in the branch

**Comment by juanpicado** [comment]
@tmkn it also fails on my local. I removed node_modules and also a new yarn lock, still broken.

**Comment by priscilawebdev** [comment]
@juanpicado a further investigation is necessary... it also fails on my local

**Comment by FilipMessa** [comment]
I will continue on this over the weekend, so will see if I will found something. 
I suppose that this is related to some module update in the `yarn.lock` file

**Comment by FilipMessa** [comment]
Hey guys, I'm sorry I didn't have time to work on this issue because of family stuff and also I changing job, so I didn't have time to continue on this but things around me going to settle. So I will be able to continue on this.

**Comment by juanpicado** [comment]
Thanks  for the feedback, hopefully things get better and no worries 😉, this can wait.

**Comment by juanpicado** [comment]
Ok, this is outdated and I think does not make much sense at this stage. Please open a new PR if someone wants to make progress on this.

**Pull request closed without merging**